x-pack/plugins/event_log/server/es/cluster_client_adapter.ts

```diff
@@ -39,18 +41,21 @@ export interface QueryEventsBySavedObjectResult {
   data: IValidatedEvent[];
 }
 
-export class ClusterClientAdapter {
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type AliasAny = any;
+
+export class ClusterClientAdapter<TDoc extends { body: AliasAny; index: string } = Doc> {
```

Comment:
We should chat with @banderror with regards to the changes he's making in `event_log` in support of [rule monitoring](https://github.com/elastic/kibana/pull/94143). I know he was looking to expose a scoped client as well, so we may have some useful overlap between these efforts.

Reply:
I haven't tried, but my hope was that we could also write rule execution events to the same indices, as I expect the mappings to be similar.